pkg/kubelet/kubelet_test.go

```diff
@@ -1407,6 +1407,109 @@ func TestCreateMirrorPod(t *testing.T) {
 	}
 }
 
+func TestPodStartDurationCountMetric(t *testing.T) {
+	metrics.Register()
+
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	// Pod with a pending status.
+	pod1 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID: "uid1",
+			Annotations: map[string]string{
+				kubetypes.ConfigFirstSeenAnnotationKey: kubetypes.NewTimestamp().GetString(),
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "container1"},
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodPending,
+		},
+	}
+	// Pod with a running status.
+	pod2 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID: "uid2",
+			Annotations: map[string]string{
+				kubetypes.ConfigFirstSeenAnnotationKey: kubetypes.NewTimestamp().GetString(),
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "container2"},
+			},
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodRunning,
+		},
+	}
+	// Pod1 container's status
+	pod1Status := &kubecontainer.PodStatus{
+		ID: pod1.UID,
+		ContainerStatuses: []*kubecontainer.Status{
+			{
+				Name:  "container1",
+				State: kubecontainer.ContainerStateCreated,
+			},
+		},
+	}
+	// Pod2 container's status
+	pod2Status := &kubecontainer.PodStatus{
+		ID: pod2.UID,
+		ContainerStatuses: []*kubecontainer.Status{
+			{
+				Name:  "container2",
+				State: kubecontainer.ContainerStateRunning,
+			},
+		},
+	}
+
+	// In the first sync process, pod1 is pending.
+	_, err := kubelet.SyncPod(context.Background(), kubetypes.SyncPodSync, pod1, nil, pod1Status)
```

Comment:
It looks to me like you are relying on this `SyncPod` call to populate the status manager so that we cover the `ok && existingStatus.Phase == v1.PodPending` case later. I think this is more confusing than it needs to be. Instead, each case in the table driven test should have a different pod, and whenever we need to test the case where `ok` is true, we populate the status manager directly using `kl.statusManager.SetPodStatus`. This will also reduce how much each test case depends on the previous one.